internal/controller/numaflowcontrollerrollout_controller.go

```diff
@@ -321,8 +329,35 @@ func (r *NumaflowControllerRolloutReconciler) getResourceOperations() (kubeUtil.
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *NumaflowControllerRolloutReconciler) SetupWithManager(mgr ctrl.Manager) error {
+
 	return ctrl.NewControllerManagedBy(mgr).
 		// Reconcile NumaflowControllerRollouts when there's been a Generation changed (i.e. Spec change)
 		For(&apiv1.NumaflowControllerRollout{}).WithEventFilter(predicate.GenerationChangedPredicate{}).
 		Complete(r)
+
+	/*
+	   controller, err := runtimecontroller.New(ControllerNumaflowControllerRollout, mgr, runtimecontroller.Options{Reconciler: r})
+
+	   	if err != nil {
+	   		return err
+	   	}
+
+	   // Watch NumaflowControllerRollouts
+
+	   	if err := controller.Watch(source.Kind(mgr.GetCache(), &apiv1.NumaflowControllerRollout{}), &handler.EnqueueRequestForObject{}, predicate.GenerationChangedPredicate{}); err != nil {
+	   		return err
+	   	}
+
+	   // Watch Deployments of numaflow-controller
```

Comment:
Do we need to watch all resources deployed for numaflow-controller, like config maps?

Reply:
Oh, yeah, you're probably right. I was just thinking about resources that we require health status for - like the numaflow-controller is Unhealthy - and we want to make sure to incorporate that. But for auto-healing, we need to be notified of it.